packages/web-components/src/components/carousel/carousel.ts

```diff
@@ -111,6 +114,39 @@ class DDSCarousel extends LitElement {
     }
   }
 
+  /**
+   * Handles card focus throughout pages.
+   *
+   * @param event The event.
+   */
+  @HostListener('shadowRoot:focusin')
+  // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
+  private _handleFocus = async ({ target, relatedTarget }: FocusEvent) => {
+    const currentContains = target !== this && this.contains(target as DDSCard);
+    const oldNotContains = target !== this && !this.contains(relatedTarget as DDSCard);
+    const currentCardIndex = Array.from(this.children).indexOf(target as HTMLElement);
+
+    // keep current page if tabbing back into the carousel after previously moving pages
+    if (currentContains && oldNotContains && currentCardIndex === 0) {
+      (this.children[this.start] as HTMLElement).focus();
+      return;
+    }
+
+    if (currentContains) {
+      // going forwards, change page depending on card index
+      if (currentCardIndex >= this.start + this.pageSize) {
+        const nextStart = currentCardIndex - (currentCardIndex % this.pageSize);
+        const pageOffset = this.start % this.pageSize;
+
+        this.start = nextStart + pageOffset;
```

Comment:
Probably you meant `this.start = currentCardIndex`? (Though there are some edge cases to consider)

Reply:
`nextStart` calculates the correct `this.start` if the user focuses on a random card (and show its containing page)
`pageOffset` adds any remainder to the calculation if needed, usually it's just 0 (adds `2` in the previous scenario) 

`this.start = currentCardIndex` only works if we tab forward but doesn't address the "random card focus" case.